### PR TITLE
SOC-1975 | Canonical link for Forum + adding acceptance tests for Discussion

### PR DESCRIPTION
**Pull request by bkoval**
## Links

* http://wikia-inc.atlassian.net/browse/SOC-1975

## Description

New pull request, needed to close https://github.com/Wikia/mercury/pull/2937

## Reviewers

@Wikia/social-frontend

### CODE REVIEW

File: front/main/app/routes/discussion/base.js [hunk 1]
```diff
@@ -18,7 +18,13 @@ export default Ember.Route.extend(
 		 */
 		setDynamicHeadTags(model, data = {}) {
 			data.documentTitle = 'Discussions';
-			data.canonical = `${Ember.get(Mercury, 'wiki.basePath')}${window.location.pathname}`;
+
+			// We do not want to set a canonical for pages other than first
+			if (this.get('controller.page') === 1) {
```
Comment (latata) [inline front/main/app/routes/discussion/base.js]: Why do you check it in base route instead of forum route?

**Comment by bkoval** [comment]
@rogatty I would like to leave double quotes for our fixtures, since otherwise it would be problematic to convert these from JSON (double quotes) to JS objects (singles where needed, rest without quotes) each time we add something new. What do you think?

File: front/main/mirage/config.js [hunk 2]
```diff
@@ -3,6 +3,7 @@
  */
 export default function () {
 	this.passthrough('/write-blanket-coverage');
+	this.passthrough('https://localhost/**');
```
Comment (rogatty) [inline front/main/mirage/config.js]: How about a const for `https://localhost/`?

**Comment by rogatty** [comment]
@bkoval would JSON files work here? I think they should.

File: front/main/app/routes/discussion/base.js [hunk 3]
```diff
@@ -18,7 +18,9 @@ export default Ember.Route.extend(
 		 */
 		setDynamicHeadTags(model, data = {}) {
 			data.documentTitle = 'Discussions';
-			data.canonical = `${Ember.get(Mercury, 'wiki.basePath')}${window.location.pathname}`;
+			if (data.canonical === undefined) {
```
Comment (mleczey) [inline front/main/app/routes/discussion/base.js]: I assume that you want to execute this code when `canonical` is only `undefined` and not when is `null`?

**Comment by bkoval** [comment]
@rogatty We'll try it next time. I just copied JSON here and exported it from fixtures since valid JSON is a valid JS object :)

File: front/main/app/routes/discussion/base.js [hunk 4]
```diff
@@ -17,8 +17,13 @@ export default Ember.Route.extend(
 		 * @returns {void}
 		 */
 		setDynamicHeadTags(model, data = {}) {
+			const shouldSetDefaultCanonical = data.canonical === undefined;
```
Comment (rogatty) [inline front/main/app/routes/discussion/base.js]: @bkoval you forgot about something? :wink: It's a shame that linter didn't shout at this unused variable.

**Pull request merged**